### DISCUSSION TITLE
Restrictions on airframe modeling

### DISCUSSION THREAD

**Discussion by jonyMarino** [JSBSim-Team/jsbsim#514] [Q&A]
In the world of UAVs, which are of my interest, we see very avant-garde designs, totally new forms that are very far from the classic airplane.
My question:
Are there restrictions on which airframes are modelable in JSBSim?
I assume that JSBSim has "low-level" ways of modeling, which will apply to almost any model. If so, what are the easily modelable types of airframes?

**Comment by seanmcleod** [comment]
No particular restrictions I can think of off-hand. At the end of the day you get to define a set of orthogonal forces and moments which are summed up for each axis, plus propulsion forces with a given position and angle etc. and then the equations of motion are applied etc.

You are free to decide how you calculate the forces and moments.

**Reply by jonyMarino** [reply]
Are there not parameters of what the wing or tail of an airplane is like that could not correspond to certain airframes?

**Reply by seanmcleod** [reply]
But you don't specify wing and tail details, for the aerodynamic forces and moments you simply specify the set of forces and moments for each axis.

For example in the JSBSim reference manual there is an example of modeling a ball being shot, this is the sum total of it's forces and moments. No references to wings or tails, or any assumptions internally in JSBSim about them.

```xml
<aerodynamics>
   <axis name="DRAG">
      <function name="aero/coefficient/CD">
         <description>Drag</description>
         <product>
            <property>aero/qbar-psf</property>
            <property>metrics/Sw-sqft</property>
            <value>0.0001</value>
         </product>
      </function>
   </axis>
</aerodynamics>
```

Take a look at some of the aircraft model files to get an idea of how the forces and moments are specified.

**Reply by jonyMarino** [reply]
I understand.
Are there some fixed-wings to which the wing and/or tail parameters would not be suitable?

**Reply by seanmcleod** [reply]
> Are there some fixed-wings to which the wing and/or tail parameters would not be suitable?

Don't really understand the question. What wing/tail parameters are you referring to?

Take a look at the reference manual - https://jsbsim.sourceforge.net/JSBSimReferenceManual.pdf

Page 109 has the simple ball example, and page 112 expands on that to add a parachute which demonstrates the use of the external reactions capability.

**Reply by jonyMarino** [reply]
Sorry, @seanmcleod, if I have been very brief.
I understand that for any aircraft, we can define a set of forces and moments.
But for airplanes, I see that we can use metrics and properties like these:
metrics:
wingarea
wingspan
chord
htailarea
htailarm
vtailarea
vtailarm
properties:
aero / alpha-wing-rad
Then a lot related to the flight controller.

I understand that using them is a facility to model a fixed-wing (the x8 model made by Alexander uses them). That is why I ask to what extent they are usable for different types of airframes. Are there designs that would make these metrics and properties unsuitable for modeling?
I'd already been playing with the ball and the parachute 😆

**Comment by seanmcleod** [comment]
> Are there designs that would make these metrics and properties unsuitable for modeling?

@jonyMarino you're reading too much into these. Just because they exist and are available to be set it doesn't mean your model needs to make use of them. Take the ball example you're looking at, it sets pretty much all of them to 0 and doesn't use any except the `wingarea` via `<property>metrics/Sw-sqft</property>`, and even that isn't required, you could specify the reference area directly as a number in the drag formula etc.

JSBSim has been used to model regular fixed wing aircraft, helicopters, quadcopters, rockets, lighter than air vehicles and even a ball.

**Reply by jonyMarino** [reply]
@seanmcleod thanks for your guidance

**Comment by andgi** [comment]
On Thu, 11 Nov 2021 at 21:51, Sean McLeod ***@***.***> wrote:
> JSBSim has been used to model regular fixed wing aircraft, helicopters,
quadcopters, rockets, lighter than air vehicles and even a ball.

And also cars, boats, a submarine and a tank... :)
See some of that on my GitHub page. Of course, some of that is maybe not
that realistic.

Cheers,
Anders

**Reply by jonyMarino** [reply]
Good work @andgi! Is there no place where people post their models?
